Gems/ROS2/Code/Include/ROS2/Lidar/LidarRaycasterBus.h

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/Component/EntityId.h>
+#include <AzCore/EBus/EBus.h>
+#include <AzCore/Math/Transform.h>
+#include <AzCore/Math/Vector3.h>
+
+namespace ROS2
+{
+    class LidarRaycasterRequests
+    {
+    public:
+        AZ_RTTI(LidarRaycasterRequests, "{253a02c8-b6cb-493c-b16f-012ccf9db226}");
+        virtual ~LidarRaycasterRequests() = default;
+
+        //! Configures ray orientations.
+        //! @param orientations Vector of orientations as Euler angles in radians. Each ray direction is computed by transforming a unit
+        //! vector in the positive z direction first by the y, next by the z axis. The x axis is currently not included in calculations.
+        virtual void ConfigureRayOrientations(const AZStd::vector<AZ::Vector3>& orientations) = 0;
+
+        //! Configures ray maximum travel distance.
+        //! @param range Ray range in meters.
+        virtual void ConfigureRayRange(float range) = 0;
+
+        //! Schedules a raycast that originates from the point described by the lidarTransform.
+        //! @param lidarTransform Current transform from global to lidar reference frame.
+        //! @return Results of the raycast in form of coordinates in 3D space.
+        //! The returned vector size can be anything between zero and size of directions. No hits further than distance will be reported.
+        virtual AZStd::vector<AZ::Vector3> PerformRaycast(const AZ::Transform& lidarTransform) = 0;
+
+        //! Configures ray Gaussian Noise parameters.
```

Comment:
These comments are excellent, super detailed and helpful for the reader, great to see! 👍

Reply:
Thanks a lot!